### PR TITLE
[2.0] Add simple bulk entity publishing benchmark

### PR DESCRIPTION
**Pull request by strawgate**
Fixes: https://github.com/legrego/homeassistant-elasticsearch/issues/310

Related to: https://github.com/legrego/homeassistant-elasticsearch/issues/278

### CODE REVIEW

File: tests/test_es_doc_publisher.py [hunk 1]
```diff
@@ -636,6 +636,46 @@ async def test_state_to_bulk_action_via_initialized_publisher(
...
+    async def test_publishing_benchmark(
+        self,
+        hass: HomeAssistant,
+        state,
+        initialized_publisher: DocumentPublisher,
+        es_aioclient_mock: AiohttpClientMocker,
+    ):
+        """Benchmark entity publishing."""
+
+        # If processing 500 entities takes more than 1s we have a significant performance degration
```
Comment (legrego) [inline tests/test_es_doc_publisher.py:657]: How did you arrive at the 500 / 1s benchmark? I worry could miss significant regressions, as it still passes on my machine when I increase it by nearly 3 orders of magnitude (500 -> 100,000).

Comment (strawgate) [reply tests/test_es_doc_publisher.py:657]: I wanted to have something first and make sure it plays nice in CI, doesn't flap, etc but I'm happy to bump it.

My plan was to follow-up later probably using the official pytest benchmark library or something.

Maybe ill parametrize it and have it test like 50, 500, 5000, and 50,000 and snapshot the duration and just make sure it's within some% of snapshot?

Comment (legrego) [reply tests/test_es_doc_publisher.py:657]: > I wanted to have something first and make sure it plays nice in CI, doesn't flap, etc but I'm happy to bump it.
> 
> My plan was to follow-up later probably using the official pytest benchmark library or something.

I'm good with this approach.

> Maybe ill parametrize it and have it test like 50, 500, 5000, and 50,000 and snapshot the duration and just make sure it's within some% of snapshot?

I don't think it's worth the complexity of rolling our own if you plan to use something like the pytest benchmark library in a follow-up

**Review by legrego — APPROVED** [review]
LGTM with the understanding that we will follow-up with more formalized benchmarking solution in the future.

**Comment by strawgate** [comment]
@legrego also I think the benchmark runs like 100x slower in the debugger w/ debug logging enabled. It takes like 5s for it to publish the state changes in my env so even though the publish is fast the test takes forevery with even 5k states

**Pull request merged**